src/store/actions.ts

```diff
@@ -96,8 +97,9 @@ export const actions = {
   },
 
   applyEffects(context: Context, changes: Change[]) {
-    const values = applyEffects({ ...(context.state.game?.values as BaseParams) }, changes)
-    const game = { ...context.state.game, values } as Game
+    const effectedContext = { dispatch: context.dispatch, values: context.state.game!.values }
```

Comment:
You do not make a copy of `values` here, anymore. I'm not sure when/if it was needed and if that changed.

Reply:
As far as I know, Vuex replaces the top level attributes in its state by accessor functions for reactivity. One should not change objects directly, but put instead a new object into the store by committing a mutation.
Actually, the `applyEffects` action does that beginning in line 102: create a new `game` object which is then committed to the store.
It should imo be no problem to use "deep" attributes of the original object for the new object, because reactivity is not affected by that. But, nevertheless, it is not good style to change it in-place how it is done here - I will change that and make a deep copy of the values.

Reply:
Ok. Some day i will try to understand, how the reactivity is done in JS by Vue(x) and others. From my naive point of view, the old object is needed intact to detect changes. That it is possible to modify the object seems to make this very fragile. Therefore, I guess it is more involved than that. Possibly, `context.state` provides us with a private copy of the current state?